### ISSUE TITLE
Docker Support

### ISSUE THREAD

**Issue by Anduin2017** [EdiWang/Moonglade#257]
Hey!

I want to run this project in my K8S cluster. For my expectation: Only a few clicks shall start a docker instance of Moonglade.

But it seems that this project doesn't. It was hard to install.

Now it is based on .NET Core and full cross-platform technology. Why don't publish it to docker hub so anyone with K8S can run it easily?

**Comment by EdiWang** [comment]
Blocked by not having SQL Server localdb on docker.

**Comment by Anduin2017** [comment]
Just use the complete version of SQL Server.

**Comment by EdiWang** [comment]
@Anduin2017 It's like killing a fly with an RPG.

**Comment by Anduin2017** [comment]
Solutions:

* Use SQLite instead of SQL Server in docker
* Use the complete version of SQL Server
* Allow the user to configure the connection string from the web during the first startup. Ask him which database he prefers and ask him for the connection string.

**Comment by EdiWang** [comment]
Initial support for Docker is done in #344 

But the blog setup with that image will blow up on the first run. This issue is under 996 investigation.

**Comment by EdiWang** [comment]
After PR #355, most path issues are fixed, but favicon still 404, it is likely to be another issue where docker doesn't have permission to write into some directories.

![image](https://user-images.githubusercontent.com/3304703/77998327-80517500-7363-11ea-833e-c782aed25f4f.png)

**Comment by EdiWang** [comment]
Favicon blow up log:
```
2020-03-31T12:02:53.405108068Z       Error generating favicons.
2020-03-31T12:02:53.405115468Z System.TypeInitializationException: The type initializer for 'Gdip' threw an exception.
2020-03-31T12:02:53.405359877Z  ---> System.DllNotFoundException: Unable to load shared library 'libgdiplus' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: liblibgdiplus: cannot open shared object file: No such file or directory
2020-03-31T12:02:53.405375177Z    at System.Drawing.SafeNativeMethods.Gdip.GdiplusStartup(IntPtr& token, StartupInput& input, StartupOutput& output)
2020-03-31T12:02:53.405390778Z    at System.Drawing.SafeNativeMethods.Gdip..cctor()
2020-03-31T12:02:53.405397878Z    --- End of inner exception stack trace ---
2020-03-31T12:02:53.405402778Z    at System.Drawing.SafeNativeMethods.Gdip.GdipLoadImageFromDelegate_linux(StreamGetHeaderDelegate getHeader, StreamGetBytesDelegate getBytes, StreamPutBytesDelegate putBytes, StreamSeekDelegate doSeek, StreamCloseDelegate close, StreamSizeDelegate size, IntPtr& image)
2020-03-31T12:02:53.405408179Z    at System.Drawing.Image.InitializeFromStream(Stream stream)
2020-03-31T12:02:53.405412979Z    at System.Drawing.Bitmap..ctor(Stream stream)
2020-03-31T12:02:53.405418279Z    at Moonglade.Web.FaviconGenerator.FileSystemFaviconGenerator.ResizeImage(Stream originImageStream, Int32 toWidth, Int32 toHeight) in /src/Moonglade.Web.FaviconGenerator/FaviconGenerator.cs:line 93
2020-03-31T12:02:53.405488582Z    at Moonglade.Web.FaviconGenerator.FileSystemFaviconGenerator.<GenerateIcons>g__WriteImageFiles|0_0(String prefix, IEnumerable`1 sizes, <>c__DisplayClass0_0& ) in /src/Moonglade.Web.FaviconGenerator/FaviconGenerator.cs:line 41
2020-03-31T12:02:53.405559884Z    at Moonglade.Web.FaviconGenerator.FileSystemFaviconGenerator.GenerateIcons(String sourceImagePath, String directory) in /src/Moonglade.Web.FaviconGenerator/FaviconGenerator.cs:line 54
2020-03-31T12:02:53.405570284Z    at Moonglade.Web.Startup.GenerateFavicons(IHostEnvironment env) in /src/Moonglade.Web/Startup.cs:line 302
```

**Comment by EdiWang** [comment]
All happy path tests are completed. Project "Moonglade" is finally ready to run on Docker! 

"That's one small step for man, one giant leap for mankind."
